packages/ui-components/src/components/Pagination/Pagination.test.js

```diff
@@ -84,6 +100,96 @@ describe("Pagination", () => {
     expect(handlePressNext).toHaveBeenCalledTimes(1)
   })
 
+  test("fires onPressNext handler with undefined currentPage and undefinded totalPages - passed when Next button is clicked", () => {
+    const handlePressNext = jest.fn()
+    render(
+      <Pagination
+        variant="number"
+        currentPage={undefined}
+        totalPages={undefined}
+        data-testid="my-pagination"
+        onPressNext={handlePressNext}
+      />
+    )
+    act(() => {
+      screen.getByRole("button", { name: "Next Page" }).click()
+    })
+    expect(handlePressNext).toHaveBeenCalledTimes(1)
+    expect(screen.getByTestId("my-pagination")).toHaveTextContent("")
+  })
+
+  test("fires onPressNext handler with undefined currentPages", () => {
+    const handlePressNext = jest.fn()
+    render(
+      <Pagination
+        variant="number"
+        currentPage={undefined}
+        totalPages={8}
+        data-testid="my-pagination"
+        onPressNext={handlePressNext}
+      />
+    )
+    act(() => {
+      screen.getByRole("button", { name: "Next Page" }).click()
+    })
+    expect(handlePressNext).toHaveBeenCalledTimes(1)
+    expect(screen.getByTestId("my-pagination")).toHaveTextContent("")
+  })
+
+  test("fires onPressNext handler with undefinded totalPages", () => {
+    const handlePressNext = jest.fn()
+    render(
+      <Pagination
+        variant="number"
+        currentPage={6}
+        totalPages={undefined}
+        data-testid="my-pagination"
+        onPressNext={handlePressNext}
+      />
+    )
+    act(() => {
+      screen.getByRole("button", { name: "Next Page" }).click()
+    })
+    expect(handlePressNext).toHaveBeenCalledTimes(1)
+    expect(screen.getByTestId("my-pagination")).toHaveTextContent("6")
+  })
+
+  test("fires onPressNext handler with higher currentPage than totalPages", () => {
```

Comment:
This test checks that the onPressNext handler is not fired in this case, so the test description should reflect this.
```suggestion
  test("does not fire onPressNext handler with higher currentPage than totalPages", () => {
```